src/modules/cpu/kernel/spectrogram.hpp

```diff
@@ -0,0 +1,239 @@
+                            *windowOutputTemp++ = windowFn[t] * srcPtrTemp[inIdx];
+                        else
+                            *windowOutputTemp++ = 0;
+                    }
+                }
+            }
+            else
+            {
+                Rpp32f *srcPtrWindowTemp = srcPtrTemp + windowStart;
+                Rpp32f *windowFnTemp = windowFn;
+                Rpp32s t = 0;
+                for (; t < alignedWindowLength; t += 8)
+                {
+                    __m256 pSrc, pWindowFn;
+                    pSrc = _mm256_loadu_ps(srcPtrWindowTemp);
+                    pWindowFn = _mm256_loadu_ps(windowFnTemp);
+                    pSrc = _mm256_mul_ps(pSrc, pWindowFn);
+                    _mm256_storeu_ps(windowOutputTemp, pSrc);
+                    srcPtrWindowTemp += 8;
+                    windowFnTemp += 8;
+                    windowOutputTemp += 8;
+                }
+                for (; t < windowLength; t++)
+                    *windowOutputTemp++ = (*windowFnTemp++) * (*srcPtrWindowTemp++);
+            }
+        }
+
+        // Generate FFT output
+        ffts_plan_t *p;
+        if(useRealImpl)
+            p = ffts_init_1d_real(nfft, FFTS_FORWARD);
+        else
+            p = ffts_init_1d(nfft, FFTS_FORWARD);
+
+        if (!p)
+        {
+            printf("FFT Plan is unsupported. Exiting the code\n");
+            exit(0);
+        }
+
+        // Set temporary buffers to 0
+        Rpp32f FFTS_ALIGN(32) *fftInBuf = (Rpp32f *)_mm_malloc(fftInSize * sizeof(Rpp32f), 32); // ffts requires 32-byte aligned memory
+        Rpp32f FFTS_ALIGN(32) *fftOutBuf = (Rpp32f *)_mm_malloc(fftOutSize * sizeof(Rpp32f), 32); // ffts requires 32-byte aligned memory
+
+        for (Rpp32s w = 0; w < numWindows; w++)
+        {
+            Rpp32f *dstPtrBinTemp = dstPtrTemp + (w * hStride);
+            Rpp32f *windowOutputTemp = windowOutput + (w * nfft);
+            for(int k = 0; k < fftInSize; k++)
+                fftInBuf[k] = 0.0f;
+
+            for(int k = 0; k < fftOutSize; k++)
+                fftOutBuf[k] = 0.0f;
+
+            Rpp32s inWindowStart = windowLength < nfft ? (nfft - windowLength) / 2 : 0;
+            // Copy the window input to fftInBuf
+            if (useRealImpl)
+            {
+                for (int i = 0; i < windowLength; i++)
+                    fftInBuf[inWindowStart + i] = windowOutputTemp[i];
+            }
+            else
+            {
+                for (int i = 0; i < windowLength; i++)
+                {
+                    Rpp32s off = 2 * (inWindowStart + i);
+                    fftInBuf[off] = windowOutputTemp[i];
+                    fftInBuf[off + 1] = 0.0f;
+                }
+            }
+
+            ffts_execute(p, fftInBuf, fftOutBuf);
+            auto *complexFft = reinterpret_cast<std::complex<Rpp32f> *>(fftOutBuf);
+            for (int i = 0; i < numBins; i++)
+            {
+                if (vertical)
```

Comment:
please remove the vertial/horizontal check outside the loop for better code

Reply:
Modified this, PR with fix will be pushed in soon.